apps/web/layouts/checkout.vue

```diff
@@ -32,22 +32,30 @@ import { SfIconArrowBack, SfLoaderCircular } from '@storefront-ui/vue';
 import type { CheckoutLayoutProps } from './types';
 
 const localePath = useLocalePath();
-const { t } = useI18n();
+const { t, locale, defaultLocale, availableLocales } = useI18n();
 const router = useRouter();
 const { isAuthorized } = useCustomer();
 const { data: cart, loading: isLoading } = useCart();
 const { setInitialData } = useInitialSetup();
 const viewport = useViewport();
 const { heading, backLabelMobile, backLabelDesktop } = defineProps<CheckoutLayoutProps>();
+
 const goToPreviousRoute = () => {
   const backPath = router.options.history.state?.back;
-
   if (isAuthorized.value && backPath === paths.guestLogin) {
     router.go(-2);
     return;
   }
-
+  const backLocaleFromPath = String(backPath).split('/')[1];
```

Comment:
refactor: https://github.com/plentymarkets/plentyshop-pwa/pull/1667 created a pr that shows how this method could refactored, if you agree with the refactor you can merge the pr into your pr.

key improvements are improved readability in the for the checkout layout, the method and usage of best practice by returning navigateTo